src/app/services/register-service.ts

```diff
@@ -0,0 +1,32 @@
+import { HttpClient, HttpErrorResponse, HttpHeaders, HttpResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, throwError } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { User } from '../models/user';
+import { catchError } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RegisterService {
+
+  constructor(private httpClient: HttpClient) { 
+
+  }
+
+  register(user: User): Observable<User>{
+    return this.httpClient.post<User>(environment.apiUrl + "users", user, {
```

Comment:
It would be nice to use a wrapper over the Angular HttpClient service that handles the requests, builds the url with path params and/or query params, set the request body...

Reply:
Here I tried to make a wrapper as generic as I could, I am open to suggestions if it is not that right.